src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/ExplicitConversionSymbolReferenceFinder.cs

```diff
@@ -43,20 +46,18 @@ protected override bool CanFind(IMethodSymbol symbol)
 
         var underlyingNamedType = GetUnderlyingNamedType(symbol.ReturnType);
         Contract.ThrowIfNull(underlyingNamedType);
-        var documentsWithName = await FindDocumentsAsync(project, documents, cancellationToken, underlyingNamedType.Name).ConfigureAwait(false);
-        var documentsWithType = await FindDocumentsAsync(project, documents, underlyingNamedType.SpecialType.ToPredefinedType(), cancellationToken).ConfigureAwait(false);
 
-        using var _ = ArrayBuilder<Document>.GetInstance(out var result);
+        using var _ = PooledHashSet<Document>.GetInstance(out var result);
+        await FindDocumentsAsync(project, documents, static (doc, result) => result.Add(doc), result, cancellationToken, underlyingNamedType.Name).ConfigureAwait(false);
+        await FindDocumentsAsync(project, documents, underlyingNamedType.SpecialType.ToPredefinedType(), static (doc, result) => result.Add(doc), result, cancellationToken).ConfigureAwait(false);
```

Comment:
nit.  ont thing i've been thinking about is the repetition of things like `static (doc, result) => result.Add(doc)`.  Consider a protected static in the topmost type like `Action<...> StandardAddCallback` and reference those here.  (may not work due to genercs).

Reply:
Went ahead and made that change (I like the idea, but these were the only ones I could find).